File: Challenges/Javascript/domainType.js
function domainType(domains) {
    let map = {
        com: "commercial",
        org: "organization",
        net: "network",
        info: "information"
    };
    
    return domains
    .map(e => map[/[a-z]{3,4}$/g.exec(e)])
}

File: Challenges/Java/DomainType.java
final class DomainType {
	
	String[] domainType(String[] domains) {
	    Map<String, String> dMap = new HashMap<>() {
	        {
	            put("com", "commercial");
	            put("org", "organization");
	            put("net", "network");
	            put("info", "information");
	        }
	    };
	    
	    return Stream.<String>of(domains).map(e -> {
	        Matcher m =  Pattern.compile("(?<=\\.)[a-z]+$").matcher(e);
	        return m.find() ? m.group() : "";
	    })
	    .map(e -> dMap.get(e))
	    .toArray(String[]::new);
	}
}